ci/main.go

```diff
@@ -71,10 +71,8 @@ func build(ctx context.Context) error {
 
 	// initialize image fron dockerfile
 	// mount a host directory in the container at the '/workspaces/parigot' path
-	img := client.Container().
-		Build(dockerDir).
-		WithDirectory(
-			"/workspaces/parigot",
+	img := dockerDir.DockerBuild().
+		WithDirectory("/workspaces/parigot",
 			client.Host().Directory("."),
 			dagger.ContainerWithDirectoryOpts{Exclude: []string{".devcontainer/", "ci/", "build/", "g/"}},
 		).
```

Comment:
I'm not totally sure what the point is of the `WithDirectoryOps` but there is a lot of files in `g/` that are dependencies for other parts of the system.  If that's not what you are doing this for, you can ignore this comment.

Reply:
- `WithDirectory` will mount the source code directory on the host at the `/workspaces/parigot/` mount point in the container
- `dagger.ContainerWithDirectoryOpts{Exclude: []string{".devcontainer/", "ci/", "build/", "g/"}}` aims to exclude these files from the mount. 

Files in `g/` will be generated at build time and I don't think they need to be mounted.